onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,578 @@
+// Copyright(C) 2019 Intel Corporation
+// Licensed under the MIT License
+
+#include <stddef.h>
+#include <algorithm>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "core/common/common.h"
+#include "core/graph/graph_viewer.h"
+#include "core/framework/compute_capability.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/session/onnxruntime_cxx_api.h"
+
+#include "openvino_execution_provider.h"
+#include "core/graph/model.h"
+#include "openvino_graph.h"
+
+#include "core/util/protobuf_parsing_utils.h"
+namespace onnxruntime {
+
+constexpr const char* OpenVINO = "OpenVINO";
+
+OpenVINOExecutionProvider::OpenVINOExecutionProvider(OpenVINOExecutionProviderInfo& info)
+    : IExecutionProvider{onnxruntime::kOpenVINOExecutionProvider} {
+  (void)info;
+
+  DeviceAllocatorRegistrationInfo device_info({OrtMemTypeDefault, [](int) { return std::make_unique<CPUAllocator>(std::make_unique<OrtAllocatorInfo>(OPENVINO, OrtDeviceAllocator, 0, OrtMemTypeDefault)); }, std::numeric_limits<size_t>::max()});
+  InsertAllocator(CreateAllocator(device_info));
+}
+
+static ONNX_NAMESPACE::ModelProto GetModelProtoFromFusedNode(const onnxruntime::GraphViewer& graph_viewer) {
+  ONNX_NAMESPACE::ModelProto model_proto;
+  auto graph_proto = model_proto.mutable_graph();
+
+  for (const auto& node : graph_viewer.Nodes()) {
+    node.ToProto(*(graph_proto->add_node()));
+  }
+
+  for (const auto& input : graph_viewer.GetInputs()) {
+    auto valueInfoProto = graph_proto->add_input();
+    *valueInfoProto = input->ToProto();
+  }
+
+  for (const auto& output : graph_viewer.GetOutputs()) {
+    auto valueInfoProto = graph_proto->add_output();
+    *valueInfoProto = output->ToProto();
+  }
+
+  for (const auto& initializer : graph_viewer.GetAllInitializedTensors()) {
+    graph_proto->add_initializer()->CopyFrom(*initializer.second);
+  }
+
+  auto opset = model_proto.add_opset_import();
+  opset->set_domain(kOnnxDomain);
+  opset->set_version(graph_viewer.DomainToVersionMap().at(kOnnxDomain));
+  model_proto.set_ir_version(ONNX_NAMESPACE::Version::IR_VERSION);
+
+  return model_proto;
+}
+
+//Checks whether the node is supported by OpenVINO
+bool isOpSupported(std::string name){
```

Comment:
shouldn't opset version also be taken into consideration?
see  https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/ngraph/ngraph_execution_provider.cc#L290

Reply:
I guess the same explanation as in https://github.com/Microsoft/onnxruntime/pull/935#discussion_r294027066 applies to this too, right?

Reply:
I will add more details to that comment.